app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -99,7 +99,16 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
                 class_name = "middleware_deployment";
                 break;
             case "MiddlewareServer":
-                class_name = "middleware_server";
+                switch (d.item.product) {
+                    case "Hawkular":
+                        class_name = "vendor-hawkular";
```

Comment:
Wrt server inheritance - You are right here. I would want to defer that implementation change  a bit until we have those new kinds of servers in Hawkular and can better tell what we really need there.